content/en/blog/_posts/2025-01-14-devices-failure-handling/index.md

```diff
@@ -0,0 +1,409 @@
+typical one is to wrap a main executable in a container by some orchestrator.
+And this orchestrator will be able to restart the main executable whenever the
+job needs to be restarted because some other pod has failed.
+
+Solutions like this are very fragile and elaborate. They are often worth the
+money saved comparing to a regular JobSet delete/recreate cycle when used in
+large training jobs. Making these solutions less fragile and more streamlined
+by developing new hooks and extension points in Kubernetes will make it
+easy to apply to smaller jobs, benefiting everybody.
+
+### Failure modes: device degradation
+
+Not all device failures are terminal for the overall workload or batch job.
+As the hardware stack gets more and more
+complex, misconfiguration on one of the hardware stack layers, or driver
+failures, may result in devices that are functional, but lagging on performance.
+One device that is lagging behind can slow down the whole training job.
+
+We see reports of such cases more and more often. Kubernetes has no way to
+express this type of failures today and since it is the newest type of failure
+mode, there is not much of a best practice offered by hardware vendors for
+detection and third party tooling for remediation of these situations.
+
+Typically, these failures are detected based on observed workload
+characteristics. For example, the expected speed of AI/ML training steps on
+particular hardware.
+
+## Roadmap
+
+As outlined in a section above, Kubernetes offers a lot of extension points
+which are used to implement various DIY solutions. The space of AI/ML is
+developing very fast, with changing requirements and usage patterns. SIG Node is
+taking a measured approach of enabling more extension points to implement the
+workload-specific scenarios over introduction of new semantics to support
+specific scenarios. This means prioritizing making information about failures
+readily available over implementing automatic remediations for those failures
+that might only be suitable for a subset of workloads.
+
+This approach ensures there are no drastic changes for workload handling which
+may break existing well-oiled DIY solutions, or experiences with the existing
+more traditional workloads.
+
+Many error handling techniques used today work for AI/ML, but are very
+expensive. SIG Node will invest in extension points to make those cheaper, with
+the understanding that the price cutting for AI/ML is critical.
+
+The following is the set of specific investments we envision for various failure
+modes.
+
+### Roadmap for failure modes: K8s infrastructure
+
+The area of Kubernetes infrastructure is the easiest to understand and very
+important to make right for the upcoming transition from Device Plugins to DRA.
+SIG Node is tracking many work items in this area, most notably the following:
+
+* **Done:** [integrate kubelet with the systemd watchdog · Issue
```

Comment:
nit: probably just list all of them? Saying `Done` might imply others are not done, which could be false as time flies.

```suggestion
* [integrate kubelet with the systemd watchdog · Issue
```